File: core/LongestNiceSubstring/LongestNiceSubstring.py
def longestNiceSubstring(s: str) -> str:
    length = len(s)
    max_pos, max_len = 0, 0
    for i in range(length):
        lower, upper = 0, 0
        for j in range(i, length):
            if s[j].islower():
                lower |= 1 << (ord(s[j]) - ord("a"))
            else:
                upper |= 1 << (ord(s[j]) - ord("A"))
            if lower == upper and j - i + 1 > max_len:
                max_len = j - i + 1
                max_pos = i

    return s[max_pos:max_pos + max_len]

File: core/LongestNiceSubstring/LongestNiceSubstring.java
package com.dbc;

public class LongestNiceSubstring {
    public String longestNiceSubstring(String s) {
        int start = 0, maxLen = 0;
        for (int i = 0; i < s.length(); i++) {
            int lower = 0, upper = 0;
            for (int j = i; j < s.length(); j++) {
                if (s.charAt(j) <= 90) {
                    lower |= 1 << (s.charAt(j) - 'a');
                } else {
                    upper |= 1 << (s.charAt(j) - 'A');
                }
                if (lower == upper && j - i + 1 > maxLen) {
                    start = i;
                    maxLen = j - i + 1;
                }
            }
        }
        return s.substring(start, start + maxLen);
    }
}
